tap_twilio/streams.py

```diff
@@ -453,10 +452,90 @@
             },
         },
     },
+    # pylint: disable=line-too-long
+    # Reference: https://www.twilio.com/docs/chat/rest/service-resource#read-multiple-service-resources
+    'services': {
+        'api_url': 'https://chat.twilio.com',
+        'api_version': 'v2',
+        'path': 'Services',
+        'data_key': 'services',
+        'key_properties': ['sid'],
+        'replication_method': 'INCREMENTAL',
+        'replication_keys': ['date_updated'],
+        'params': {},
+        'pagination': 'root',
+        'children': {
+            # pylint: disable=line-too-long
+            # Reference: https://www.twilio.com/docs/chat/rest/role-resource#read-multiple-role-resources
+            'roles': {
+                'api_url': 'https://chat.twilio.com',
+                'api_version': 'v2',
+                'path': 'Services/{ParentId}/Channels/{ParentId}/Roles',
+                'data_key': 'roles',
+                'key_properties': ['sid'],
+                'replication_method': 'INCREMENTAL',
+                'replication_keys': ['date_updated'],
+                'params': {},
+                'pagination': 'root'
+            },
+            # pylint: disable=line-too-long
+            # Reference: https://www.twilio.com/docs/chat/rest/channel-resource#read-multiple-channel-resources
+            'chat_channels': {
+                'api_url': 'https://chat.twilio.com',
+                'api_version': 'v2',
+                'path': 'Services/{ParentId}/Channels',
+                'data_key': 'channels',
+                'key_properties': ['sid'],
+                'replication_method': 'INCREMENTAL',
+                'replication_keys': ['date_updated'],
+                'params': {},
+                'pagination': 'root',
+                'children': {
+                    'members': {
+                        'api_url': 'https://chat.twilio.com',
+                        'api_version': 'v2',
+                        'path': 'Services/{ParentId}/Channels/{ParentId}/Members',
+                        'data_key': 'members',
+                        'key_properties': ['sid'],
+                        'replication_method': 'FULL_TABLE',
+                        'params': {},
+                        'pagination': 'root'
+                    },
+                    # pylint: disable=line-too-long
+                    # Reference: https://www.twilio.com/docs/chat/rest/message-resource#read-multiple-message-resources
+                    'chat_messages': {
+                        'api_url': 'https://chat.twilio.com',
+                        'api_version': 'v2',
+                        'path': 'Services/{ParentId}/Channels/{ParentId}/Messages',
+                        'data_key': 'messages',
+                        'key_properties': ['sid'],
+                        'replication_method': 'FULL_TABLE',
```

Comment:
We have an issue with this replicating every message.
If we set it to `INCREMENTAL`, a bookmark will be stored for the resource itself, not per the parent resource (chat_channels). That means if we find 100 chat_channels, and for example the 5th one has the newest chat_message until now, we will store that in the bookmark. If the next chat_channel has for example 10 messages, and the first 5 messages are older than the stored bookmark, the others are newer, only the newer ones will be stored. If we stumble upon a chat_channel which is still new, we haven't processed it, but all of its message are older than a previous chat_channel's messages, none of the messages will be stored.

On the other hand, if we go with `FULL_TABLE`, we will synch all the messages from all the channels. In December, we had 35k channels (=chats), if we take 50 as the average number of messages (25 from agent, 25 from customer), that's 1.75 million entries. 

None of these solution seems feasible, so I think we'll need to find a solution how to synch children resources for only only parents which are new.

Reply:
We decided not to solve this now. We are able to synch these resources, too, but since it would be too much data, we can decide not to synch them.
Added a warning message to the `README`.

Reply:
we'll need to think about how to make it incremental.
We potentially want to sync multiple times a day and having `FULL_TABLE` I think we can't schedule to run this tap more than once a day.

Do we need to worry about any API rate limit that will block using the API endpoing after a certain number of requests? If I understand correctly we're about to send millions of requests to the same endpoint every day.